src/Compilers/CSharp/Test/Symbol/Compilation/GetUnusedImportDirectivesTests.cs

```diff
@@ -375,5 +375,61 @@ public void UnusedUsingScript()
                 // using System;
                 Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using System;"));
         }
+
+        [Fact, WorkItem(18348, "https://github.com/dotnet/roslyn/issues/18348")]
+        public void IncorrectUnusedUsingWhenAttributeOnParameter_01()
+        {
+            var source1 =
+@"using System.Runtime.InteropServices;
+
+partial class Program
+{
+    partial void M([Out] [In] ref int x) { }
+}";
+            var source2 =
+@"partial class Program
+{
+    partial void M(ref int x);
+}";
+            var comp = CreateStandardCompilation(new[] { source1, source2 });
+            var tree = comp.SyntaxTrees[0];
+            //comp.VerifyDiagnostics(); // doing this first hides the symptoms of the bug
+            var model = comp.GetSemanticModel(tree);
+
+            // There should be no diagnostics.
+            model.GetDiagnostics().Verify(
```

Comment:
>model.GetDiagnostics().Verify( [](start = 12, length = 30)

Please also add a unit-test that asserts diagnostics requested exclusively for source1 syntax tree.

Reply:
That is precisely what this line does.